File: 0067-add-binary/0067-add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        # Binary addition is used here 

        # ans = ""
        # carry = 0
        # a,b = a[::-1], b[::-1]

        # for i in range(max(len(a), len(b))):
        #     digitA = int(a[i]) if i < len(a) else 0
        #     digitB = int(b[i]) if i < len(b) else 0

        #     total = digitA+digitB+carry
        #     ans = str(total%2) + ans
        #     carry = total // 2

        # if carry: ans = "1"+ans
        # return ans


        # I wrote fucking both code, because i didnt knew method bin exists
        return bin(int(a,2)+int(b,2))[2:]
        

File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        // return Integer.toBinaryString(Integer.parseInt(a, 2)+Integer.parseInt(b, 2));

        StringBuilder ans = new StringBuilder();

        int i = a.length() - 1;
        int j = b.length() - 1;
        int carry = 0;

        while (i >= 0 || j >= 0 || carry > 0) {

            int sum = carry;

            if (i >= 0) {
                sum += a.charAt(i) - '0';
                i--;
            }

            if (j >= 0) {
                sum += b.charAt(j) - '0';
                j--;
            }

            ans.append(sum % 2);
            carry = sum / 2;
        }

        return ans.reverse().toString();
    }
}